bin/p2-bootstrap/bootstrap.go

```diff
@@ -68,17 +89,37 @@ func InstallConsul(consulPod *pods.Pod, consulManifest *pods.PodManifest) error
 	if err != nil || !ok {
 		return util.Errorf("Can't launch Consul, aborting: %s", err)
 	}
-	time.Sleep(time.Second * 10)
 	return nil
 }
 
-func RegisterBaseAgentInConsul(agentManifest *pods.PodManifest) error {
+func VerifyConsulUp() error {
+	store := kp.NewStore(kp.Options{})
+	consulIsUp := make(chan struct{})
+	go func() {
+		for {
+			time.Sleep(200 * time.Millisecond)
+			err := store.Ping()
+			if err == nil {
+				consulIsUp <- struct{}{}
+				return
+			}
+		}
+	}()
+	select {
+	case <-time.After(10 * time.Second):
```

Comment:
Let me take this for a spin in universe, I'm really eager to see if it eliminates my awful "ping consul in ruby using raw net::http" code.

As an aside for future improvements, it might be useful to configure the timeout or set it to zero to skip it altogether.